client/EditContributionView.js

```diff
@@ -113,6 +144,43 @@ class EditContributionView extends Component {
             });
             this.setState({contributionData: contrib});
         }
+
+        this.handleSwitchChange = (event) => {
+            let contributionData = this.state.contributionData;
+            let publishedList = this.state.publishedList;
+            let publishedSwitch = this.state.publishedSwitch;
+            let approvalSwitch = this.state.approvalSwitch;
+            if (event.target.name === 'publishedSwitch') {
+                if (!this.props.admin) {
+                    console.log("Attempt to publish without admin credentials!");
+                    return;
+                } else if (event.target.checked) {
+                    contributionData.status = "published";
+                    contributionData.approval = "approved";
+                    approvalSwitch = false;
+                    publishedSwitch = true;
+                    publishedList[contributionData.ref.id] = 'true';
+                } else {
+                    contributionData.status = "unpublished";
+                    contributionData.approval = "not requested";
+                    approvalSwitch = false;
+                    publishedSwitch = false;
+                    publishedList[contributionData.ref.id] = 'false';
+                }
+            } else if (event.target.name === 'approvalSwitch') {
+                contributionData.approval = (event.target.checked) ? "pending" : "not requested";
+                approvalSwitch = event.target.checked;
+            } else {
+                console.log("Something called this function...but what??");
+                return;
+            }
+            this.setState({
+                contributionData: contributionData,
+                publishedSwitch: publishedSwitch,
+                approvalSwitch: approvalSwitch,
+                publishedList: publishedList,
```

Comment:
can be replaced with firebase update call (like how you used get in App.js) so App.js can be replaced with bindDoc. This will reduce hits to database since we only get once in root component and pass down via props, and update once in relevant child component